lib/streamlit/elements/number_input.py

```diff
@@ -289,13 +289,20 @@ def _number_input(
         # Ensure that the value matches arguments' types.
         all_ints = int_value and int_args
 
-        if (min_value and min_value > value) or (max_value and max_value < value):
-            raise StreamlitAPIException(
-                "The default `value` of %(value)s "
-                "must lie between the `min_value` of %(min)s "
-                "and the `max_value` of %(max)s, inclusively."
-                % {"value": value, "min": min_value, "max": max_value}
-            )
+        if min_value is not None and value is not None:
+            if min_value > value:
+                raise StreamlitAPIException(
+                    "The default `value` of %(value)s "
+                    "must be greater than or equal to the `min_value` of %(min)s "
+                    % {"value": value, "min": min_value}
+                )
```

Comment:
Nit: we tend to use [f-strings](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings) for simple interpolations like this:

```python
f"The default `value` {value} must be greater than or equal to the `min_value` {min_value}"
```